### PR TITLE
Readme

### LINKED ISSUE #22: Start off the project documentation by writing a Getting Started guide
We should also decide as part of this goal how we're going to provide project documentation. Options include using the Wiki or a Github site maintained in a branch of this repo or in a subdirectory.

### PR DESCRIPTION
**Pull request by tengpengli**
**Commit Message:**

Changes README for the present codes.
1) Gives a snippet to import our repository
2) Uses a tiny example to show the usage.
3) Creates docs/cc_deps.md for the macro in cc_deps.bzl.
4) Creates docs/common.md for the rules in common.bzl.

**Additional Description:**
**Testing:** N/A
**Docs Changes:**
README.md, docs/cc_deps.md, docs/common.md
**Fixes #22**

Signed-off-by: tengpeng <tengpeng.li2020@gmail.com>

	modified:   README.md
	new file:   docs/cc_deps.md
	new file:   docs/common.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,4 +4,60 @@ This repository contains [Bazel](https://bazel.build/)
 [Starlark extensions](https://docs.bazel.build/versions/master/skylark/concepts.html)
 for defining fuzz tests in Bazel projects. 
 
-**This is not an officially supported Google product.**
\ No newline at end of file
+**This is not an officially supported Google product.**
+
+## Getting started
+
+To import rules_fuzzing in your project, you first need to add codes below to your `WORKSPACE` file:
```
Comment (stefanbucur) [inline README.md:11]: To import the *fuzzing rules* in your project...

Comment (stefanbucur) [inline README.md:11]: s/codes/the snippet/

File: README.md [hunk 2]
```diff
@@ -4,4 +4,60 @@ This repository contains [Bazel](https://bazel.build/)
...
+
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+        name = "rules_fuzzing",
+        sha256 = "8d98c444600ce25082b320a3364150806109fe33620a4e1c5bb856861591acb9",
+        strip_prefix = "bazel-rules-fuzzing-8520424714e827ff0360283e3237e35c8723d931",
```
Comment (stefanbucur) [inline README.md:18]: Just curious: Why do we need the `strip_prefix` attribute here?

File: docs/cc_deps.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<a name="#cc_fuzz_test"></a>
+
+## cc_fuzz_test
+
+<pre>
+cc_fuzz_test(<a href="#cc_fuzz_test-name">name</a>, <a href="#cc_fuzz_test-corpus">corpus</a>, <a href="#cc_fuzz_test-kwargs">**kwargs</a>)
+</pre>
```
Comment (stefanbucur) [inline docs/cc_deps.md:7]: Why use HTML here? Is there something that Markdown can't provide?

File: docs/common.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<a name="#fuzzing_launcher"></a>
+
+## fuzzing_launcher
+
+<pre>
+fuzzing_launcher(<a href="#fuzzing_launcher-name">name</a>, <a href="#fuzzing_launcher-corpus">corpus</a>, <a href="#fuzzing_launcher-is_regression">is_regression</a>, <a href="#fuzzing_launcher-target">target</a>)
+</pre>
```
Comment (stefanbucur) [inline docs/common.md:7]: Same here.

File: docs/cc_deps.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+
+**PARAMETERS**
+
+
+| Name  | Description | Default Value |
+| :-------------: | :-------------: | :-------------: |
+| name |  A unique name for this target.   |  none |
+| corpus |  A list containing corpus files.   |  <code>None</code> |
+| kwargs |  Keyword arguments.   |  none |
```
Comment (stefanbucur) [inline docs/cc_deps.md:25]: I'm wondering if we can auto-format these tables. @asraa is there something Envoy is using that we could use here too?

File: docs/cc_deps.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<a name="#cc_fuzz_test"></a>
```
Comment (stefanbucur) [inline docs/cc_deps.md:1]: Tengpeng, I believe the standard is to include API documentation directly in the source code, as Starlark docstrings.

File: README.md [hunk 7]
```diff
@@ -4,4 +4,60 @@ This repository contains [Bazel](https://bazel.build/)
...
+load("@rules_fuzzing//fuzzing:cc_deps.bzl", "cc_fuzz_test")
+
+cc_fuzz_test(
+    name = "fuzz_test",
+    srcs = ["fuzz_test.cc"],
+    corpus = ["corpus_1.txt"] + glob(["corpus_dir/**"],
+)
+```
+
+With the config setting in `.bazelrc`, you then can run the fuzz test above using command
```
Comment (stefanbucur) [inline README.md:47]: Maybe expand here how users should create their `.bazelrc` file?

File: README.md (hunk 2 above)
Comment (tengpengli) [reply README.md:18]: After checking https://docs.bazel.build/versions/master/repo/http.html#http_archive I think we can leave this option to the user.

File: docs/cc_deps.md (hunk 6 above)
Comment (tengpengli) [reply docs/cc_deps.md:1]: Just to clarify, do you mean we don't need `cc_deps.md` and `common.md` to list those rules, instead, just the docstrings in the source code?

File: docs/cc_deps.md (hunk 5 above)
Comment (asraa) [reply docs/cc_deps.md:25]: Not that I know of, we don't format markdown files directly

File: docs/cc_deps.md (hunk 6 above)
Comment (asraa) [reply docs/cc_deps.md:1]: When included, recommend specifying what `corpus` expects as srcs

File: docs/common.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+<a name="#fuzzing_dictionary"></a>
+
+## fuzzing_dictionary
+
+<pre>
+fuzzing_dictionary(<a href="#fuzzing_dictionary-name">name</a>, <a href="#fuzzing_dictionary-dicts">dicts</a>, <a href="#fuzzing_dictionary-output">output</a>)
+</pre>
+
+
+This rule validates the fuzzing dictionaries and output a merged dictionary.
```
Comment (asraa) [inline docs/common.md:55]: What happens when validation fails? Suggest example error message or reference to what validations are done (link to libfuzzer)

File: docs/cc_deps.md (hunk 5 above)
Comment (tengpengli) [reply docs/cc_deps.md:25]: I was using [stardoc](https://github.com/bazelbuild/stardoc) to generate the tables in `common.md` and here. I am wondering if there is any format standard to follow?

**Comment by tengpengli** [comment]
@asraa @stefanbucur PTAL!

I need this base version of README so that in my next PR I can change the documentation based on this version.

**Pull request merged**